services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -222,22 +218,24 @@ postMLSCommitBundleToLocalConv qusr c conn bundle ctype lConvOrSubId = do
     note (mlsProtocolError "Unsupported ciphersuite") $
       cipherSuiteTag bundle.groupInfo.value.groupContext.cipherSuite
 
-  -- when a user tries to join any mls conversation while being legalholded
+  -- when a user tries to join any mls conversation while being under legalhold
   -- they receive a 409 stating that mls and legalhold are incompatible
   case qusr `RelativeTo` lConvOrSubId of
     Local luid ->
       when (isNothing convOrSub.mlsMeta.cnvmlsActiveData) do
-        usrTeams <- getUserTeams (tUnqualified luid)
+        usrTeams <- TeamStore.getUserTeams (tUnqualified luid)
         for_ usrTeams \tid -> do
           -- this would only return 'Left' if the team member did vanish directly in the process of this
           -- request or if the legalhold state was somehow inconsistent. We can safely assume that this
           -- should be a server error
-          Right resp <- runError @(Tagged TeamMemberNotFound ()) $ getUserStatus luid tid (tUnqualified luid)
-          case resp.ulhsrStatus of
-            UserLegalHoldPending -> throwS @MLSLegalholdIncompatible
-            UserLegalHoldEnabled -> throwS @MLSLegalholdIncompatible
-            UserLegalHoldDisabled -> pure ()
-            UserLegalHoldNoConsent -> pure ()
+          resp <- runError @(Tagged TeamMemberNotFound ()) $ getUserStatus luid tid (tUnqualified luid)
+          case resp of
+            Left _ -> throw $ InternalErrorWithDescription "Server error. Team member must have vanished during the legal hold check"
```

Comment:
i guess in the spirit of `Nothing -> pure ()` above, we could throw the more confusing but less loud MLSLegalholdIncompatible here?

but i am leaning towards @MangoIV and think internal errors that are supposed to be caught by tests shouldn't take up so much space in the code.

...  and in the comments of the related PRs, for that matter...  :-)  so i'll shut up.

Reply:
`MLSLegalholdIncompatible` is a semantic error, the internal error here is an internal error, right? State changed from under us midway through a function. :)

Reply:
> but i am leaning towards @MangoIV and think internal errors that are supposed to be caught by tests shouldn't take up so much space in the code.

this was exactly the intention my idea was that they should be explained in the code for whenever they occur and then not take much space at runtime or at compile time as they’re not part of the usual usage. :)